holopy/fitting.py

```diff
@@ -78,32 +77,32 @@ def __new__(self,guess=None, limit=None, name=None, **kwargs):
         setattr(out, 'kwargs',kwargs)
         return out
 
-class ComplexParameter(HoloPyObject):
+class ComplexParameter():
     def __new__(self, real, imag, name=None):
         fit_warning('hp.inference.prior.ComplexPrior', 'ComplexParameter')
         if isinstance(real, Prior) or isinstance(imag, Prior):
             return ComplexPrior(real, imag, name)
         else:
             return real + 1.0j * imag
 
-class ParameterizedObject(HoloPyObject):
+class ParameterizedObject():
     def __new__(self, obj):
         fit_warning('Scatterer', 'ParameterizedObject')
         setattr(obj, 'make_from', obj.from_parameters)
         return obj
 
-class Parametrization(HoloPyObject):
+class Parametrization():
     def __new__(self, scatclass, pars):
         fit_warning('Scatterer', 'Parametrization')
         return ParameterizedObject(scatclass(*pars))
 
 
-class limit_overlaps(HoloPyObject):
+class limit_overlaps():
```

Comment:
This entire file exists only to preserve backwards compatibility and will be deleted in an upcoming version of HoloPy. We need to keep the namespace exactly as it was, but I could make this a function instead of a class, since it behaves like one anyway. I ran into problems doing that with some other classes because they couldn't be used to instantiate objects from saved data values if they were functions. I don't know whether that specific issue would be a problem for `limit_overlaps` but it could be.

To further complicate things, `limit_overlaps` is a function (not a class) on the current master branch. It was changed to a class very early in the 3.3 development cycle for compatibility with parallel computation. The original name was maintained then against PEP8 to avoid breaking things because it was anticipated as a minor 3.2.2 release, and not as 3.3. So any potential problems with loading HoloPy objects that contain `limit_overlap` objects would be limited to users of the pre-release 3.3 branch.

I think it's safest to keep the version currently in the PR as the file will soon be deleted anyway, but there doesn't seem to be a big risk in making this a function if you prefer.

Reply:
OK to keep as is.  Mark with a brief comment covering what you wrote above.  Rest is ready to merge.